eglib/src/gstr.c

```diff
@@ -38,6 +38,12 @@
 
 #include <errno.h>
 
+#if defined (_POSIX_C_SOURCE) && defined (_GNU_SOURCE)
+#if (_POSIX_C_SOURCE >= 200112L) && !_GNU_SOURCE
+#define HAVE_STRERROR_R_XSI
```

Comment:
HAVE_xxxx macros should only be defined by configure.

It's ok to check it here, but use a different name prefix, like USE_xxx.

Additionally, there should be a comment on why this is being checked.